### ISSUE TITLE
TypeError: fmin() got an unexpected keyword argument 'rseed'

### ISSUE THREAD

**Issue by jonhilgart22** [maxpumperla/hyperas#82]
Hi there,

When trying to run this code, I receive the following error.

from __future__ import print_function

from hyperopt import Trials, STATUS_OK, tpe
from keras.datasets import mnist
from keras.layers.core import Dense, Dropout, Activation
from keras.models import Sequential
from keras.utils import np_utils
import numpy as np
from hyperas import optim
from keras.models import model_from_json
from keras.models import Sequential
from keras.layers.core import Dense, Dropout, Activation, Flatten
from keras.layers import LSTM
from keras.optimizers import SGD , Adam
import tensorflow as tf
import keras.backend as K
from hyperas.distributions import choice, uniform, conditional
from keras.layers.normalization import BatchNormalization
__author__ = 'JOnathan Hilgart'

def data():
    """
    Data providing function:

    This function is separated from model() so that hyperopt
    won't reload data for each evaluation run.
    """
    import numpy as np
    x = np.load('training_x')
    x = x.reshape(1,50001,2)
    y = np.load('training_y')
    y = y.reshape(1,50001, 9)
    x_train = x[:, :25000,:]
    y_train = y[:, :25000,:]
    x_test = x[:, 25001:,:]
    y_test = y[:, 25001:,:]
    return x_train, y_train, x_test, y_test

def model(x_train, y_train, x_test, y_test):
    """
    Model providing function:

    Create Keras model with double curly brackets dropped-in as needed.
    Return value has to be a valid python dictionary with two customary keys:
        - loss: Specify a numeric evaluation metric to be minimized
        - status: Just use STATUS_OK and see hyperopt documentation if not feasible
    The last one is optional, though recommended, namely:
        - model: specify the model just created so that we can later use it again.
    """
    model_lstm = Sequential()
    model_lstm .add(LSTM({{choice([64, 126, 256, 512, 1024])}}, dropout={{uniform(0, .5)}},
                         batch_input_shape=(1,x_train.shape[1], 2),
                     recurrent_dropout={{uniform(0, .5)}},return_sequences = True))
    model_lstm.add(BatchNormalization())

    if conditional({{choice(['one','two','three', 'four'])}}) == 'one':
        pass
    elif conditional({{choice(['one','two','three', 'four'])}}) == 'two':
        model_lstm .add(LSTM({{choice([64, 126, 256, 512, 1024])}}, dropout={{uniform(0, .5)}},
                     recurrent_dropout={{uniform(0, .5)}},
                     return_sequences = True))
        model_lstm.add(BatchNormalization())
    elif conditional({{choice(['one','two','three', 'four'])}}) == 'three':
        model_lstm .add(LSTM({{choice([64, 126, 256, 512, 1024])}}, dropout={{uniform(0, .5)}},
                     recurrent_dropout={{uniform(0, .5)}},
                     return_sequences = True))
        model_lstm.add(BatchNormalization())
        model_lstm.add(Dense({{choice([126, 256, 512, 1024])}},
                             name='dense_one'))
        model_lstm.add(BatchNormalization())
        model_lstm.add(Activation('relu'))
    elif conditional({{choice(['one','two','three', 'four'])}}) == 'four':
        model_lstm .add(LSTM({{choice([64, 126, 256, 512, 1024])}}, dropout={{uniform(0, .5)}},
                     recurrent_dropout={{uniform(0, .5)}},
                     return_sequences = True))
        model_lstm.add(BatchNormalization())
        model_lstm.add(Dense({{choice([126, 256, 512, 1024])}},
                             name='dense_one'))
        model_lstm.add(BatchNormalization())
        model_lstm.add(Activation('relu'))
        model_lstm.add(Dense({{choice([126, 256, 512, 1024])}}, activation='relu',
                             name='dense_one'))
        model_lstm.add(BatchNormalization())
        model_lstm.add(Activation('relu'))

    model_lstm .add(Dense(9, activation='linear',name='dense_output'))
    model_lstm .compile(loss='mean_squared_error', optimizer={{choice(['adam','sgd'])}})
    model_lstm.summary()

    model_lstm.fit(x_train, y_train,
              batch_size=1,
              epochs=1,
              verbose=1,
              validation_data=(x_test, y_test))
    score, acc = model_lstm.evaluate(x_test, y_test, verbose=0)
    print('Test accuracy:', acc)
    return {'loss': -acc, 'status': STATUS_OK, 'model': model_mlp}

if __name__ == '__main__':
    import gc; gc.collect()

    best_run, best_model = optim.minimize(model=model,
                                          data=data,
                                          algo=tpe.suggest,
                                          max_evals=1,
                                          trials=Trials())
    X_train, Y_train, X_test, Y_test = data()
    print("Evalutation of best performing model:")
    print(best_model.evaluate(X_test, Y_test))
    print("Best performing model chosen hyper-parameters:")
    print(best_run)

---------------------------

Traceback (most recent call last):
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperas/optim.py", line 83, in base_minimizer
    rseed=rseed)
TypeError: fmin() got an unexpected keyword argument 'rseed'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "hyperparameter_optimization_lstm.py", line 112, in <module>
    trials=Trials())
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperas/optim.py", line 42, in minimize
    notebook_name=notebook_name, verbose=verbose)
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperas/optim.py", line 90, in base_minimizer
    rstate=np.random.RandomState(rseed))
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/fmin.py", line 307, in fmin
    return_argmin=return_argmin,
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/base.py", line 635, in fmin
    return_argmin=return_argmin)
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/fmin.py", line 320, in fmin
    rval.exhaust()
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/fmin.py", line 199, in exhaust
    self.run(self.max_evals - n_done, block_until_done=self.async)
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/fmin.py", line 173, in run
    self.serial_evaluate()
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/fmin.py", line 92, in serial_evaluate
    result = self.domain.evaluate(spec, ctrl)
  File "/Users/jonathanhilgart/anaconda/envs/dl/lib/python3.6/site-packages/hyperopt/base.py", line 840, in evaluate
    rval = self.fn(pyll_rval)
  File "/Users/jonathanhilgart/galvanize-projects/deep_learning/src/models-DQN/temp_model.py", line 172, in keras_fmin_fnct
TypeError: 'numpy.float64' object is not iterable

-----------------------------

I've installed hyperas from pip and from git with no luck.

Thanks!

**Comment by jonhilgart22** [comment]
Got it.

I was trying to return score, acc from  

score, acc = model_lstm.evaluate(x_test, y_test, verbose=0)

but, you should only return acc

acc = model_lstm.evaluate(x_test, y_test, verbose=0)

**Comment by Pr0duktiv** [comment]
I had the exact same issue, and your solution was not working (quite the contrast, after fixing the issue I had, your solution threw another exception).

To fix my exception of fmin() getting rseed, I looked at the core file (~/lib/python3.5/site-packages/hyperas) and checked the line in which the error occured (83). It passes on the argument rseed=rseed to fmin(), so far so good. After checking the fmin() function however, I found out that this function does not take that argument. After manually removing rseed=rseed from 
```
best_run = fmin(keras_fmin_fnct,
                        space=get_space(),
                        algo=algo,
                        max_evals=max_evals,
                        trials=trials,
                        rseed=rseed)
```

my code worked just fine.
